src/dllist.c

```diff
@@ -739,8 +739,17 @@ static PyObject* dllist_str(DLListObject* self)
     return dllist_to_string(self, PyObject_Str);
 }
 
+/* NOTE - THIS FUNCTION DOES NOT WORK!!
+*
+*   dllist([1, 5, 9]) has the SAME hash as dllist([5, 1, 9])
+*     and thus it is NOT a hash function
```

Comment:
You are right that we should have chosen a better hashing function. I'll change it to something that detects simple reordering of list items.